### ISSUE TITLE
Add Support for Simulation in XY Mode

### ISSUE THREAD

**Issue by sebgrijalva** [pasqal-io/Pulser#147]
With the recently added support for `Microwave` channels and a corresponding "XY Mode" in a Pulser sequence (Refer to [this reference](https://arxiv.org/pdf/1408.1055.pdf) for some background on the physics), it remains to add the corresponding simulation support. This implies adding an XY Hamiltonian which takes as time-dependent coefficients the pulses of the sequence, and will imply several changes to the `Simulation` class:

- A  `self._interaction` or `self._sequence_mode` private attribute will allow to classify the incoming sequence depending on the chosen mode (should be related to `Sequence._in_xy`).
- The `self.samples` dictionary has to correspond to the case of 'Ising' or 'XY' interaction: In the case of the latter, since there is only one basis (`'xy'`), we only need slots for `'Global'` and `'Local'`.
- Since the sequence has been verified to have consistent basis of pulses throughout, the `write_samples()` function inside the `self._extract_samples()`method would only be slightly modified. On the other hand, the `self._build_basis_and_op_matrices()` method should now reflect the two-Rydberg level nature of the interaction , while following the existing scheme of the (actually more complex) Ising interaction. Perhaps just call these levels 'up' and 'down'?
- In the `make_interaction_term()` function of the `self._construct_hamiltonian()` method, one needs to add the proper term `0.5 * C_3 / R_ij**3`, which depends on the cube of the interatomic distance and has now a different coupling constant `C_3` (which should be taken as  `C_3 = 3.7 10**3 Mhz * µm**3`). 
- In fact, the coefficient `C_3` depends on the relative position vector of the atoms (Cf. Supplementary Material of [the Reference](https://arxiv.org/pdf/1408.1055.pdf)) as `C_3(theta) = C_3 * (1 - 3*cos(theta)**2)`. This dependance should be included by reading from the `Register` the position of the corresponding atoms.
- The `build_coeffs_ops()`function inside `self._construct_hamiltonian()`should also take into account the differently built operators.
- Finally, one must edit the `SimulationResults` class: In this case, both the `basis_name` and `meas_basis` correspond to the "XY mode" which implies `SimulationResults._dim = 2`. Thus, many of the checks and basis reduction methods are simpler, although they have to be clearly assessed.

To check things are working correctly, one could try to add a test displaying Rabi Oscillations with similar properties as that of Fig. 1(c) of the [Reference](https://arxiv.org/pdf/1408.1055.pdf), with one and then several atoms in an array and global/local pulses.

**Comment by HGSilveri** [comment]
Closing this just until #144 is merged.

**Comment by Slimane33** [comment]
Hello,
I would like to do it.

**Comment by HGSilveri** [comment]
**Status update:**

Support for XY Simulation has been successfully added and is currently residing in the `xy-simulation` branch. Since the work on this feature started, the `develop` branch has diverged to incorporate support for noisy simulations. Therefore, the changes in `xy-simulation` now need to be reconciled with those in the `develop` branch, with the goal of ultimately merging `xy-simulation` into `develop`.

**Comment by HGSilveri** [comment]
**Status update**:

The `xy-simulation` has been (apparently) successfully merged into `noisy-xy` (which was branched from `develop`). This issue can then be closed once `noisy-xy` is merged into `develop`.
